python/ray/data/read_api.py

```diff
@@ -305,18 +304,16 @@ def read_datasource(
     if ray_remote_args is None:
         ray_remote_args = {}
 
-    local_uri = False
-    paths = read_args.get("paths", None)
-    if paths and _is_local_scheme(paths):
+    if not datasource.supports_distributed_reads:
         if ray.util.client.ray.is_connected():
             raise ValueError(
-                f"The local scheme paths {paths} are not supported in Ray Client."
+                "If you're using Ray Client, Ray Data won't schedule read tasks "
+                "on the driver's node."
```

Comment:
```suggestion
                "If you're using Ray Client, Ray Data won't schedule read tasks "
                "on the driver's node."
```

This error message seems a bit off/indirect. Probably best to say something about files on the client's node not being accessible, and recommend passing a distributed filesystem path.

Spark has a good error messages [guideline](https://spark.apache.org/error-message-guidelines.html) (probably don't need to cover it all here but just for reference).

Reply:
Yeah, I wanted to avoid including file-specific details in `read_datasource`, because that'd be breaking an abstraction barriers (assuming that the datasource is file related).

I've moved the error into `FileBasedDatasource` and `ParquetDatasource`, and updated the message

Reply:
Thanks, message looks good now!